### PR TITLE
stats: Add network and http statisics

### PR DESCRIPTION
**Pull request by vadmeste**
## Description
Network: total bytes of incoming and outgoing server's data
by taking advantage of our ConnMux Read/Write wrapping

HTTP: total number of different http verbs passed in http
requests and different status codes passed in http responses.
This is counted in a new http handler.

## Motivation and Context
Partially f.i.x.e.s #3679 

## How Has This Been Tested?
Manually

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by vadmeste** [comment]
@harshavardhana, I tested with a sampe code, using atomic is more faster, thanks for pointing it out.

```
package main

import (
        "sync"
        "sync/atomic"
        "testing"
)

type safeTest struct {
        atomicCounter uint64

        mutexCounter  uint64
        mu sync.Mutex
}

// Benchmark increasing counter with mutex lock/unlock
func BenchmarkSafeMutex(b *testing.B) {
        s := &safeTest{}
        for i := 0; i < b.N; i++ {
                s.mu.Lock()
                s.mutexCounter++
                s.mu.Unlock()
        }
}

// Benchmark increasing counter with atomic Add
func BenchmarkSafeAtomic(b *testing.B) {
        s := &safeTest{}
        for i := 0; i < b.N; i++ {
                atomic.AddUint64(&s.atomicCounter, 1)
        }
}
```

Result:
```
BenchmarkSafeMutex-4            50000000                25.1 ns/op
BenchmarkSafeAtomic-4           200000000                9.19 ns/op
PASS
ok      _/tmp/test      4.053s
```

**Comment by vadmeste** [comment]
@harshavardhana, I used atomic wherever it is possible. Not possible to use atomic with map.

File: cmd/stats.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+import (
+	"sync"
+	"sync/atomic"
+)
+
+// ConnStats - Network statistics
+// Count total input/output transferred bytes during
+// the server's life.
+type ConnStats struct {
```
Comment (harshavardhana) [inline cmd/stats.go:42]: Lets actually generalize this 

```go
type uint64 counterer

func (c counterer) Inc(uint64 i) {
}

func (c counterer) Value() uint64 {
}
```

**Comment by vadmeste** [comment]
@harshavardhana, the code below shows that map is slower than atomically increasing static variables even with using a switch statement.

So the new approach looks good.

```
package main

import (
        "sync"
        "sync/atomic"
        "testing"
)

func BenchmarkUpdateMap(b *testing.B) {
        method := "PUT"
        mu := sync.Mutex{}
        m := make(map[string]uint64)

        for i := 0; i < b.N; i++ {
                mu.Lock()
                m[method]++
                mu.Unlock()
        }
}

func BenchmarkUpdateStatic(b *testing.B) {
        method := "PUT"
        staticPUT := uint64(0)
        staticOTHER := uint64(0)

        for i := 0; i < b.N; i++ {
                switch method {
                case "PUT":
                        atomic.AddUint64(&staticPUT, 1)
                case "POST":
                        atomic.AddUint64(&staticOTHER, 1)
                case "GET":
                        atomic.AddUint64(&staticOTHER, 1)
                case "HEAD":
                        atomic.AddUint64(&staticOTHER, 1)
                case "DELETE":
                        atomic.AddUint64(&staticOTHER, 1)
                }
        }
}
```
```
testing: warning: no tests to run
BenchmarkUpdateMap-4            20000000                56.5 ns/op
BenchmarkUpdateStatic-4         100000000               16.9 ns/op
PASS
ok      _/tmp/test      2.903s
```

**Pull request merged**